benchmark_utils/backbones_architecture.py

```diff
@@ -72,3 +73,25 @@ def forward(self, x, sample_weight=None):
         x = self.mlp(x)
 
         return x    # Return raw logits
+
+
+class FBCSPNet(nn.Module):
+    def __init__(self, n_chans, n_classes, input_window_samples):
+        super().__init__()
+
+        # Create the ShallowFBCSPNet
+        self.feature_layer = ShallowFBCSPNet(
+            n_chans, n_classes, input_window_samples, add_log_softmax=False, final_conv_length='auto'
+        )
+
+        # Take last layer from the ShallowFBCSPNet
+        self.final_layer = self.feature_layer.final_layer
+
+        # Replace ShallowFBCSPNet's last layer with an identity function
+        self.feature_layer.final_layer = nn.Identity()
+
+    def forward(self, x, sample_weight=None):
+        x = self.feature_layer(x)
```

Comment:
So doing that you will reduce the divergence between the output of the source and the target not between the features. You have to get the name in ShallowFBCSPNet of the feature extractor. You can check again, but I think it will be "drop". And depending of the dataset or model you have to use the good layer name. If you prefer using always the same maybe you can change the name of one layer of a model, but I don't know how.